server/text_generation_server/layers/attention/kv_cache.py

```diff
@@ -94,17 +147,33 @@ def store(
         key: torch.Tensor,
         value: torch.Tensor,
         slots: torch.Tensor,
+        kv_scales: KVScales,
     ):
         """Store the key and value at the given slots."""
 
         key_cache = self.kv_cache[0]
         value_cache = self.kv_cache[1]
 
+        if self.can_scale(kv_scales):
+            if kv_scales.key_scale_cpu != 1.0:
+                key = fp8_quantize(
+                    key.float(),
+                    scale=kv_scales.key_scale,
+                    qdtype=self.dtype,
+                    scalar=True,
+                )[0]
+            if kv_scales.value_scale_cpu != 1.0:
+                value = fp8_quantize(
+                    value.float(),
+                    scale=kv_scales.value_scale,
+                    qdtype=self.dtype,
+                    scalar=True,
+                )[0]
+
```

Comment:
Should this be only done when ATTENTION is `flashinfer`, since scales are not passed to flash decoding and paged attention yet?

Reply:
Ah yes, good catch! I added an additional condition to `can_scale` to check that `ATTENTION=="flashinfer"`.